com.dotsui.core/DotsUI.Controls/ScrollRectSystem.cs

```diff
@@ -0,0 +1,229 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+using Unity.Collections;
+using Unity.Entities;
+using Unity.Jobs;
+using DotsUI.Core;
+using Unity.Burst;
+using Unity.Mathematics;
+using Unity.Transforms;
+using UnityEngine;
+using RectTransform = DotsUI.Core.RectTransform;
+
+namespace DotsUI.Controls
+{
+    [UpdateInGroup(typeof(PostRectTransformSystemGroup))]
+    class ScrollRectSystem : JobComponentSystem
+    {
+        private EntityQuery m_ScrollRectQuery;
+
+        protected override void OnCreate()
+        {
+            m_ScrollRectQuery = GetEntityQuery(ComponentType.ReadOnly<ScrollRect>(), ComponentType.ReadWrite<WorldSpaceRect>());
+        }
+
+        struct ScrollViewTransforms
+        {
+            public RectTransform ContentTransform;
+            public RectTransform ViewportTransform;
+            public WorldSpaceRect ScrollViewRect;
+            public WorldSpaceRect ContentRect;
+            public WorldSpaceRect ViewportRect;
+
+            public RectTransform VerticalAreaTransform;
+            public RectTransform VerticalHandleTransform;
+            public RectTransform HorizontalAreaTransform;
+            public RectTransform HorizontalHandleTransform;
+        }
+
+        [BurstCompile]
+        struct UpdateScrollRectData : IJobChunk
+        {
+            [ReadOnly] public ArchetypeChunkEntityType EntityType;
+            [ReadOnly] public ArchetypeChunkComponentType<ScrollRect> ScrollRectType;
+            //public ArchetypeChunkComponentType<ScrollRectData> ScrollRectData;
+
+            [NativeDisableParallelForRestriction] public ComponentDataFromEntity<WorldSpaceRect> WorldSpaceRectFromEntity;
```

Comment:
It seems like a dangerous assumption. In the current implementation, there is a possibility that two overlapping hierarchies would be updated in different jobs. I think I'm going to manually iterate over chunks in IJob instead of parallel IJobChunk. I'll add it to TODO list. However, I'm not going to hold 0.2.0 release.